### PR TITLE
Is/submit tx worker

### PR DESCRIPTION
**Pull request by hSloan**

### CODE REVIEW

File: hydra-pay/src/HydraPay/Database/Workers.hs [hunk 1]
```diff
@@ -38,6 +38,8 @@ data PaymentChannelReq
   | PaymentChannelReq_Join Int32 Text Api.Lovelace
   | PaymentChannelReq_SpinUpHead Int32
   | PaymentChannelReq_Cleanup Int32
+  | PaymentChannelReq_Poke
```
Comment (luigy) [inline hydra-pay/src/HydraPay/Database/Workers.hs:41]: what's the role of `Poke`?

File: hydra-pay/src/HydraPay/PaymentChannel/Postgres.hs [hunk 2]
```diff
@@ -345,3 +345,13 @@ getHydraHeadStatusQ headId = do
     guard_ (head_ ^. Db.hydraHead_id ==. val_ (SqlSerial headId))
     pure head_
   pure $ maybe HydraHeadStatus_Unknown Db._hydraHead_status mHead
+
+checkAddressAvailability :: MonadBeam Postgres m => Api.AddressAny -> m Bool
+checkAddressAvailability addr = do
+  isAvail <- runSelectReturningOne $ select $ do
+    aa <- all_ $ Db.db ^. Db.db_addressAvailability
+    guard_ (aa ^. Db.addressAvailability_layer1Address ==. val_ (T.pack $ show addr))
```
Comment (luigy) [inline hydra-pay/src/HydraPay/PaymentChannel/Postgres.hs:353]: `Cardano.Api.serialiseAddress` would be a better fit here

**Review by luigy — DISMISSED** [review]

File: hydra-pay/src/HydraPay.hs [hunk 3]
```diff
@@ -251,3 +251,51 @@ handleChannelRefund state refundRequest = do
...
+
+handleSubmitTx :: MonadIO m => HydraPayState -> Text -> BS.ByteString -> m (Either Text Text)
+handleSubmitTx state l1Address tx = do
+  availability <- isAvailable l1Address
+  case availability of
+    True -> do
+      setAvailability l1Address False
+      eTx :: Either Text TxId <- submitTxCbor state tx
+      case eTx of
+        Left err -> return $ Left err
```
[the remark this answers is not in the document]
Comment (hSloan) [reply hydra-pay/src/HydraPay.hs:263]: Great catch, no there isn't. Will set availability to True when a tx error is encountered to allow retries later on.

File: hydra-pay/src/HydraPay.hs [hunk 4]
```diff
@@ -251,3 +251,51 @@ handleChannelRefund state refundRequest = do
...
+      lookupRes <- runSelectReturningList $ select $ do
+        aa <- all_ (DB.db ^. DB.db_addressAvailability)
+        guard_ (aa ^. DB.addressAvailability_layer1Address ==. val_ l1Address)
+        pure aa
+      case lookupRes of
+        [] -> return True
+        res:_ -> return $ res ^. DB.addressAvailability_isAvailable
+    setAvailability :: MonadIO m =>Text -> Bool -> m ()
+    setAvailability addr avail = DB.runBeam (_hydraPay_databaseConnectionPool state) $ do
+      lookupRes <- runSelectReturningList $ select $ do
```
[the remark this answers is not in the document]
Comment (hSloan) [reply hydra-pay/src/HydraPay.hs:285]: Slightly different, but just noticed I could abstract the query that both of these are using. Thanks a bunch!

File: hydra-pay/src/HydraPay/Database/Workers.hs (hunk 1 above)
Comment (hSloan) [reply hydra-pay/src/HydraPay/Database/Workers.hs:41]: Typo, thanks a bunch.

**Review by Yasuke — APPROVED** [review]
LGTM

**Pull request merged**